src/lib.rs

```diff
@@ -4,127 +4,170 @@ pub struct DB<T> {
 }
 
 /// An immutably borrowed subset of a DB
-///
-/// NB: (nota bene, or "take special note"): You will need to be explcit about the liftimes in this
-/// struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBView<T> {
-    entries: Vec<&T>,
+pub struct DBView<'a, T: 'a> {
+    entries: Vec<&'a T>,
 }
 
 /// An mutably borrowed subset of a DB
-///
-/// NB: You will need to be explcit about the liftimes in this struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBViewMut<T> {
-    entries: Vec<&mut T>,
+pub struct DBViewMut<'a, T: 'a> {
+    entries: Vec<&'a mut T>,
 }
 
 /// Filters a DBView using the the given predicate.
-pub fn filter_one<T, F>(view: &DBView<T>, predicate: F) -> DBView< T>
+pub fn filter_one<'a, T, F>(view: &DBView<'a, T>, predicate: F) -> DBView<'a, T>
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    view.select_where(predicate)
 }
 
 /// Filters two DBView structs using the same predicate, producing two separate results. This is
 /// the moral equivalent of doing the two filters separately.
-pub fn filter_two<T, F>(view_a: &DBView<T>,
-                        view_b: &DBView<T>,
+pub fn filter_two<'a, 'b, T, F>(view_a: &DBView<'a, T>,
+                        view_b: &DBView<'b, T>,
                         predicate: F)
-                        -> (DBView<T>, DBView<T>)
+                        -> (DBView<'a, T>, DBView<'b, T>)
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    (filter_one(view_a, &predicate), filter_one(view_b, &predicate))
 }
 
 impl<T> DB<T> {
     /// Creates a DB from the given list of entries
     pub fn new(data: Vec<T>) -> DB<T> {
-        unimplemented!()
+        DB { data: data }
     }
 
     /// Creates a new DBView containing all entries in `self` which satisfy `predicate`
-    pub fn select_where<F>(&self, predicate: F) -> DBView<T>
+    pub fn select_where<'a, F>(&'a self, predicate: F) -> DBView<'a, T>
         where F: Fn(&T) -> bool
     {
-        unimplemented!()
+        let mut selected = Vec::new();
+        for x in &self.data {
+            if predicate(x) {
+                selected.push(x);
+            }
+        }
+
+        DBView{entries: selected}
```

Comment:
This could be done using the Iterator interface. Good places to start: `Iterator::filter`, `Iterator::collect`, and`Iterator::cloned`.